Nimble/Matchers/Call.swift

```diff
@@ -0,0 +1,194 @@
+            failureMessage.postfixActual = kNilReminderString
+            return false
+        }
+        
+        let includeOption = didCallResultIncludeOptionFor(isNegationTest: isNegationTest)
+        let result = expressionValue.didCall(function: function, withArgs: arguments, atLeast: atLeast, recordedCallsDescOption: includeOption)
+        
+        if !isSuccessfulTest(result.success, isNegationTest) {
+            failureMessage.postfixMessage = descriptionOfAttemptedCall(object: expressionValue, function: function, arguments: arguments, countDescription: "at least", count: atLeast)
+            failureMessage.actualValue = result.recordedCallsDescription
+        }
+        
+        return isSuccessfulTest(result.success, isNegationTest)
+    }
+}
+
+public func call(function function: String, withArguments arguments: [Any], atMost: Int) -> FullMatcherFunc<CallRecorder> {
+    return FullMatcherFunc { expression, failureMessage, isNegationTest in
+        guard let expressionValue = try expression.evaluate() else {
+            failureMessage.postfixMessage = kCallFunc + kWithArgs + kAtMost
+            failureMessage.postfixActual = kNilReminderString
+            return false
+        }
+        
+        let includeOption = didCallResultIncludeOptionFor(isNegationTest: isNegationTest)
+        let result = expressionValue.didCall(function: function, withArgs: arguments, atMost: atMost, recordedCallsDescOption: includeOption)
+        
+        if !isSuccessfulTest(result.success, isNegationTest) {
+            failureMessage.postfixMessage = descriptionOfAttemptedCall(object: expressionValue, function: function, arguments: arguments, countDescription: "at most", count: atMost)
+            failureMessage.actualValue = result.recordedCallsDescription
+        }
+        
+        return isSuccessfulTest(result.success, isNegationTest)
+    }
+}
+
+// MARK: Private
+
+private let kCallFunc = "call function"
+private let kWithArgs = " with arguments"
+private let kCount = " count times"
+private let kAtLeast = " at least count times"
+private let kAtMost = " at most count times"
+private let kNilReminderString = " (use beNil() to match nils)"
+
+private func descriptionOfAttemptedCall(object object: Any, function: String, arguments: [Any], countDescription: String, count: Int) -> String {
+    var description = "call <\(function)> from \(object.dynamicType)"
+    
+    if !arguments.isEmpty {
+        let argumentsDescription = arguments.map{ "\($0)" }.joinWithSeparator(", ")
+        description += " with \(argumentsDescription)"
+    }
+    
+    if !countDescription.isEmpty {
+        let pluralism = count == 1 ? "" : "s"
+        description += " \(countDescription) \(count) time\(pluralism)"
+    }
+    
+    return description
+}
+
+private func isSuccessfulTest(didDoIt: Bool, _ isNegationTest: Bool) -> Bool {
+    return didDoIt && !isNegationTest || !didDoIt && isNegationTest
```

Comment:
Fixed by changing the bool check to only have to make one comparison instead of one, two, or three (depending on short-circuiting). It's probably insignificant savings, but when it comes to running a lot of tests, every little bit helps.